src/wrapper.js

```diff
@@ -20,25 +20,53 @@
 
 		// For CommonJS and CommonJS-like environments where a proper `window`
 		// is present, execute the factory and get jQuery.
-		// For environments that do not have a `window` with a `document`
+		// For environments that do not have a global `document`
 		// (such as Node.js), expose a factory as module.exports.
-		// This accentuates the need for the creation of a real `window`.
-		// e.g. var jQuery = require("jquery")(window);
-		// See ticket #14549 for more info.
-		module.exports = global.document ?
-			factory( global, true ) :
-			function( w ) {
-				if ( !w.document ) {
-					throw new Error( "jQuery requires a window with a document" );
+
+		module.exports = ( global.window && global.window.document ) ?
+			factory( global.window, true, global ) :
+			function( global, window ) {
```

Comment:
> This line is what I was talking about - you require jQuery Node.js users to pass the
> global object as the first parameter

No, they may pass either window or global for the first argument. I'll add a comment making that clear. The behavior is the same as before if you pass window. See my explanation of the various scenarios on the issues page. Most scenarios do not use the second argument (only one IIRC).

Went over that on the issue page. It's simply a convenience that goes along with the rest of the untangling. It's certainly more appropriate to have jQuery adding the properties to the global object, rather than to the `window` host object. There's no complication to the API, which remains the same for the caller, unless the pass the global object in NodeJS or use the second argument to pass a fake window in addition to the global object.